indico/modules/auth/controllers.py

```diff
@@ -435,11 +435,17 @@ class Meta:
             first_name = fields.String(required=True)
             last_name = fields.String(required=True)
             address = fields.String(load_default='')
-            affiliation = fields.String(load_default='')
+            if 'affiliation' in user_management_settings.get('mandatory_fields_account_request'):
+                affiliation = fields.String(required=True)
```

Comment:
`validate=not_empty` would be a good idea here as well (otherwise an empty string would be accepted)